matcher/config.go

```diff
@@ -85,7 +86,7 @@ func (cfg *Configuration) NewOptions(
 	instrumentOpts instrument.Options,
 ) (Options, error) {
 	// Configure rules kv store.
-	rulesStore, err := kvCluster.Store(cfg.RulesKVNamespace)
+	rulesStore, err := kvCluster.Store(cfg.RulesKVConfig.NewOptions())
```

Comment:
Do you need to validate the options here?

Reply:
The opts.Validate() is called in the Store()